4617_Sprint/9-Haoran/Demo.java

```diff
@@ -0,0 +1,57 @@
+import java.util.*;
+import java.time.*;
+import java.time.format.*;
+
+public class Demo {
+
+  private void siriMode(String input, Siri siri) {
+
+    if (input.contains("sassy") && "normal".equals(siri.getChatbotMode())) {
+      siri.setChatbotMode("sassy");
+      System.out.println("SassySiri model on.");
+    } else if ("sassy".equals(siri.getChatbotMode()) && input.contains("normal") ) {
+      siri.setChatbotMode("normal");
+      System.out.println("Normal model on.");
+    }
+  }
+
+  private void response(String input, Siri siri){
+    switch (siri.getChatbotMode()) {
+      case "sassy":
+      Siri sassySiri = new SassySiri();
```

Comment:
Since you are creating a SassySiri here it might be an idea to do so for normal Siri aswell in the other switch case
Improving consistency and being able to forgo the second param.

However the best solution to find would be using polymorphism and by the use of casting make from the siri a SassySiri